dracut/modules.d/99kiwi-lib/module-setup.sh

```diff
@@ -22,7 +22,7 @@ install() {
         e2fsck btrfsck xfs_repair \
         vgs vgchange lvextend lvcreate lvresize pvresize \
         mdadm cryptsetup dialog \
-        pv curl xz \
+        pv curl xz dolly\
```

Comment:
I would not do that. The initrd creation will fail if dolly is not installed in the rootfs. This will affect many image descriptions out there unless we add the `dolly` as a dependency to `dracut-kiwi-oem-dump` package and I think this is something we can't effort. `curl` is out there in every single distro and part of the very basic tools each distro provides and likely to be installed by default in most cases. I doubt this is the case for `dolly`.

 `dracut` provides an alternative for this kind of situations:
 
 ```
inst_multiple [-o] <file> [ <file> ...]
           installs multiple binaries and files. If executables are
           specified without a path, dracut will search the path
           PATH=/usr/sbin:/sbin:/usr/bin:/bin for the binary. If the
           option "-o" is given as the first parameter, a missing file
           does not lead to an error.
 ```
 So I'd say this lines should be changed to:
 ```bash
...
     pv curl xz dmsetup
 inst_multiple -o dolly
if [[ "$(uname -m)" =~ s390 ]];then
...
 ```
 
 This way we ensure the initrd creation does not break if dolly is not present in the system. I assume that anyone using the `dolly://` scheme in URIs will be well aware the ` dolly` binaries are required to be part of the rootfs. If someone uses `dolly://` URI without `dolly` binaries will just fail at run time with a command-not-found sort of issue and I bet build logs will also show a warning from dracut for the missing `dolly` command.

Reply:
This is interesting I thought dracut will not exit with an error if the file is missing. iirc when people configured an image to be used with lvm but did not install lvm2 the dracut call did not fail for the missing lvcreate and friends.

In any case I agree, a missing dolly should not harm the initrd creation process.

Thanks for catching this one

Reply:
Good point. Thanks for bringing this up. Its now optional.